cli/module_graph.rs

```diff
@@ -1852,6 +1851,53 @@ impl GraphBuilder {
     Ok(())
   }
 
+  /// Analyze compiler options, identifying any specifiers that need to be
+  /// resolved and added to the graph.
+  pub async fn analyze_compiler_options(
+    &mut self,
+    maybe_compiler_options: &Option<HashMap<String, Value>>,
+  ) -> Result<(), AnyError> {
+    if let Some(user_config) = maybe_compiler_options {
+      if let Some(value) = user_config.get("types") {
+        let types: Vec<String> = serde_json::from_value(value.clone())?;
+        for specifier in types {
+          if let Ok(specifier) = resolve_url_or_path(&specifier) {
+            self.insert(&specifier, false).await?;
+          }
+        }
+      }
+    }
+    Ok(())
+  }
+
+  /// Analyze a config file, identifying any specifiers that need to be resolved
+  /// and added to the graph.
+  pub async fn analyze_config_file(
```

Comment:
I presume these two methods are required to handle both strictly types compiler options coming from `ConfigFile` as well as JSON value coming from runtime compiler APIs?

Reply:
For `Deno.emit()` no, we pass a `Value` from the op when we merge it into the config.

Reply:
Correct.